syntax = "proto3";

// Not necessary for Python but should still be declared to avoid name collisions
// in the Protocol Buffers namespace and non-Python languages
package protoblog;

// Style guide prefers prefixing enum values instead of surrounding
// with an enclosing message
enum Device_Conection_State {
    DEVICE_IDLE = 0;
    DEVICE_CONNECTED= 1;
}

message AC{
    Device_Conection_State conection_state = 1;
    string state = 2;
    int32 temperature = 3;
    enum Device_Conection_State {
    DEVICE_IDLE = 0;
    DEVICE_CONNECTED= 1;
    }
}
message IrrigationSystem{
    Device_Conection_State conection_state = 1;
    string state = 2;
    int32 intensity = 3;
    enum Device_Conection_State {
    DEVICE_IDLE = 0;
    DEVICE_CONNECTED= 1;
    }
}
message lamp {
    string device_name = 1;
    Device_Conection_State conection_state = 2;
    string state = 3;

    enum Device_Conection_State {
    DEVICE_IDLE = 0;
    DEVICE_CONNECTED= 1;
    }
}

message MESSAGE{
    string msg = 6;
}